pcdsdevices/lodcm.py

```diff
@@ -60,6 +70,760 @@ def __init__(self, prefix, *args, **kwargs):
+    x2_si = FCpt(OffsetMotor, prefix='{self._prefix}:X2:OFF_Si', name='x2_si',
+                 motor_prefix='{self._m_prefix}:MON:MMS:11',
+                 add_prefix=('prefix', 'motor_prefix'), kind='normal',
+                 doc='X2 motor offset for Si [mm]')
+    y2_c = FCpt(OffsetMotor, prefix='{self._prefix}:Y2:OFF_C', name='y2_c',
+                motor_prefix='{self._m_prefix}:MON:MMS:12',
+                add_prefix=('prefix', 'motor_prefix'), kind='normal',
+                doc='Y2 motor offset for C [mm]')
+    y2_si = FCpt(OffsetMotor, prefix='{self._prefix}:Y2:OFF_Si', name='y2_si',
+                 motor_prefix='{self._m_prefix}:MON:MMS:12',
+                 add_prefix=('prefix', 'motor_prefix'), kind='normal',
+                 doc='Y2 motor offset for Si [mm]')
+    chi2_c = FCpt(OffsetMotor, prefix='{self._prefix}:CHI2:OFF_C',
+                  name='chi2_c', motor_prefix='{self._m_prefix}:MON:MMS:14',
+                  add_prefix=('prefix', 'motor_prefix'), kind='normal',
+                  doc='Chi 2 motor offset for C [deg]')
+    chi2_si = FCpt(OffsetMotor, prefix='{self._prefix}:CHI2:OFF_Si',
+                   name='chi2_si', motor_prefix='{self._m_prefix}:MON:MMS:14',
+                   add_prefix=('prefix', 'motor_prefix'), kind='normal',
+                   doc='Chi 2 motor offset for Si [deg]')
+    h2n_c = FCpt(OffsetMotor, prefix='{self._prefix}:H2N:OFF_C', name='h2n_c',
+                 motor_prefix='{self._m_prefix}:MON:MMS:15',
+                 add_prefix=('prefix', 'motor_prefix'), kind='normal',
+                 doc=' H2n motor offset for C [mm]')
+    h2n_si = FCpt(OffsetMotor, prefix='{self._prefix}:H2N:OFF_Si',
+                  name='h2n_si', motor_prefix='{self._m_prefix}:MON:MMS:15',
+                  add_prefix=('prefix', 'motor_prefix'), kind='normal',
+                  doc='H2n motor offset for Si [mm]')
+
+    tab_component_names = True
+    tab_whitelist = ['is_diamond', 'is_silicon', 'get_reflection',
+                     'get_material']
+
+    def __init__(self, prefix, *args, **kwargs):
+        self._m_prefix = ''
+        self._prefix = prefix
+        if 'XPP' in prefix:
+            self._m_prefix = 'XPP'
+        elif 'XCS' in prefix:
+            self._m_prefix = 'HFX'
+        super().__init__(prefix, *args, **kwargs)
+
+    def is_diamond(self):
+        """Check if tower 2 is with Diamond (C) material."""
+        return ((self.h2n_state.position == 'C' or
+                 self.h2n_state.position == 'OUT') and
+                self.y2_state.position == 'C' and
+                self.chi2_state.position == 'C')
+
+    def is_silicon(self):
+        """Check if tower 2 is with Silicon (Si) material."""
+        return ((self.h2n_state.position == 'Si' or
+                 self.h2n_state.position == 'OUT') and
+                self.y2_state.position == 'Si' and
+                self.chi2_state.position == 'Si')
+
+    def get_reflection(self, as_tuple=False, check=False):
```

Comment:
I got rid of the check from `get_material` - that one was more annoying and easier to get rid of.
I think in the next PR i should get rid of this one too, but I need to know what exactly we want to do here, since this check was a bit more involved

Reply:
Changed the behavior in https://github.com/pcdshub/pcdsdevices/pull/766/commits